### PR TITLE
delete a connector and its connection

### PR DESCRIPTION
**Pull request by arun3688**
### Related Issues

https://github.com/OpenModelica/OMSimulator/issues/748

### Purpose

This PR fixes the problem involved when deleting a connector , should also deletes it associated connections.

### changed files

System.cpp -- added new function "deleteConnectionsToConnector"  which deltes the connection associtated with the connector.

### CODE REVIEW

File: src/OMSimulatorLib/Component.h [hunk 1]
```diff
@@ -101,6 +101,8 @@ namespace oms
     virtual oms_status_enu_t setInteger(const ComRef& cref, int value) {return logError_NotImplemented;}
     virtual oms_status_enu_t setReal(const ComRef& cref, double value) {return logError_NotImplemented;}
 
+    virtual oms_status_enu_t deleteStartValues(const ComRef& cref) {return oms_status_ok;}
```
Comment (lochel) [inline src/OMSimulatorLib/Component.h:104]: This is just deleting a single start value at a time, right? Hence I would rename it to `deleteStartValue`.

File: src/OMSimulatorLib/System.cpp [hunk 2]
```diff
@@ -1505,7 +1506,8 @@ oms_status_enu_t oms::System::delete_(const oms::ComRef& cref)
     for (int i=0; i<busconnectors.size()-1; ++i)
       if (busconnectors[i]->getName() == front)
       {
-        deleteAllConectionsTo(front);
+        //deleteAllConectionsTo(front);
```
Comment (lochel) [inline src/OMSimulatorLib/System.cpp:1509]: Why did you remove this call?

Comment (arun3688) [reply src/OMSimulatorLib/System.cpp:1509]: i see two problems, 

1. This call actuall did not work, it did not delete the connections associated with the connector, and that is why this issue was repored
2. This call will delete all the connections assosciated with the "System or component", but that should not be the case , (e.g) we have two connection 

```
System2.C3 -> System1.C1
System2.C4 -> System1.C2
```
And now we want to delete 

`oms_delete("System2.C3")
`
The above call "deleteAllConectionsTo(front);" will delete all the connections associated with "System2" but that should not be the case , only this connection should be deleted

`System2.C3 -> System1.C1
`

File: src/OMSimulatorLib/Component.h (hunk 1 above)
Comment (arun3688) [reply src/OMSimulatorLib/Component.h]: renamed the function name to deleteStartValue see https://github.com/OpenModelica/OMSimulator/pull/758/commits/ee8d2f44ed637c3fcc93e87b8f81dbffd999b5ff

File: testsuite/OMSimulator/deleteConnector.lua [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+oms_addConnection("deleteConnector.Root.System2.C4", "deleteConnector.Root.System1.C2")
+
+src = oms_list("deleteConnector")
+print(src)
+
+-- delete connector "deleteConnector.Root.System2.C3" and associated connection
+oms_delete("deleteConnector.Root.System2.C3")
+
+-- delete components connector and its startValue
+oms_delete("deleteConnector.Root.System1.Gain.k")
```
Comment (adeas31) [inline testsuite/OMSimulator/deleteConnector.lua:36]: Are we actually allowed to delete this, since `k` is part of the FMU?

Comment (arun3688) [reply testsuite/OMSimulator/deleteConnector.lua:36]: @adeas31 , yes we are allowed to do this,  as different tools does not export all the variables related to FMUs, some tools export only inputs and outputs

File: src/OMSimulatorLib/Parameters.cpp [hunk 4]
```diff
@@ -70,6 +70,33 @@ oms_status_enu_t oms::Parameters::setBoolean(const ComRef& cref, bool value)
...
+  }
+  // booleanstartValues
+  auto boolValue = booleanStartValues.find(cref);
+  if (boolValue != booleanStartValues.end())
+  {
+    booleanStartValues.erase(boolValue);
+    return oms_status_ok;
+  }
+
+  return logError("deleteStartValue failed for : " + std::string(cref));
```
Comment (lochel) [inline src/OMSimulatorLib/Parameters.cpp:97]: This should rather be a warning, right?

Comment (arun3688) [reply src/OMSimulatorLib/Parameters.cpp:97]: you are correct, it can be a warning

File: src/OMSimulatorLib/System.cpp [hunk 5]
```diff
@@ -1628,6 +1628,16 @@ oms_status_enu_t oms::System::delete_(const oms::ComRef& cref)
...
+      if (Flags::ExportParametersInline())
+      {
+        // delete startValues associated with the Connector
+        component->second->deleteStartValue(tail);
+      }
+      else
+      {
+        // delete startValues associated with the Connector from ssv file
+        component->second->deleteStartValue(cref);
+      }
```
Comment (lochel) [inline src/OMSimulatorLib/System.cpp:1640]: Please move this to `oms::Component::deleteConnector`

Comment (arun3688) [reply src/OMSimulatorLib/System.cpp]: moved the code to `oms::Component::deleteConnector` see https://github.com/OpenModelica/OMSimulator/pull/758/commits/b669e30afeea6d53ecfb6f68d9e9e71474db8dfc

File: src/OMSimulatorLib/System.cpp [hunk 6]
```diff
@@ -1590,13 +1620,19 @@ oms_status_enu_t oms::System::delete_(const oms::ComRef& cref)
...
 
     auto component = components.find(front);
     if (component != components.end())
+    {
+      deleteAllConectionsTo(cref);
       return component->second->deleteConnector(tail);
+    }
   }
-
+  logWarning("failed to delete object \"" + std::string(getFullCref()+front) + "\"" + " because the identifier couldn't be resolved to any connector, component, system, or model");
```
Comment (lochel) [inline src/OMSimulatorLib/System.cpp:1635]: I guess it should be `std::string(getFullCref() + cref)`. `tail` doesn't necessarily need to be empty.

Comment (arun3688) [reply src/OMSimulatorLib/System.cpp]: changed the warning message https://github.com/OpenModelica/OMSimulator/pull/758/commits/6bf8b45f0071257e82813e5d62aa24321c2d47e8
